e2e/nomostest/predicates.go

```diff
@@ -578,3 +582,46 @@ func SecretMissingKey(key string) Predicate {
 		return nil
 	}
 }
+
+// NamspaceConfigHasMetadataName will check the metadata name and compare it with the expected value
+func NamspaceConfigHasMetadataName(expectedName string) Predicate {
+	return func(o client.Object) error {
+		raw := o.(*v1.NamespaceConfig).Spec.Resources[0].Versions[0].Objects[0].Raw
```

Comment:
All of these sub-indices seem a bit too specific for a generic sounding function like `NamspaceConfigHasMetadataName`. This also appears to only be used in one test, so perhaps just define the Predicate inline with the test.

ditto for `NamspaceConfigHasMetadataCreationTimestamp`

Reply:
predicates have been moved to test